drivers/adc/adc_ads1x4s0x.c

```diff
@@ -1047,18 +1112,52 @@ static int ads1x4s0x_adc_perform_read(const struct device *dev)
 		return result;
 	}
 
-	result = ads1x4s0x_read_sample(dev, data->buffer);
+#if ADS1X4S0X_HAS_24_BIT_DEV && ADS1X4S0X_HAS_16_BIT_DEV
```

Comment:
It should be possible to reduce the `#if` cases here:

	if (config->resolution == 24) {
		#if ADS1X4S0X_HAS_24_BIT_DEV
		do_something_24();
		#else
		failed
		#endif
	}

	if (config->resolution == 16) {
		#if ADS1X4S0X_HAS_16_BIT_DEV
		do_something_16();
		#else
		failed
		#endif
	}

I'm not a big fan of goto-statements, even for failure handling. But this is more of a personal preference of mine, it is rather widely used in the zephyr codebase.

Reply:
done